pkg/serverless/flush/strategy.go

```diff
@@ -7,17 +7,33 @@ package flush
 
 import (
 	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"math"
+	"math/rand"
 	"strconv"
 	"strings"
+	"sync"
 	"time"
 )
 
+const maxBackoffRetrySeconds = 5 * 60
+
 // Strategy is deciding whether the data should be flushed or not at the given moment.
 type Strategy interface {
 	String() string
 	ShouldFlush(moment Moment, t time.Time) bool
+	Failure(t time.Time)
+	Success()
+}
+
+type retryState struct {
+	lastFail time.Time
+	retries  uint64
+	lock     sync.Mutex
 }
 
+var globalRetryState = retryState{}
```

Comment:
I'm concerned this might need to be a pointer to a `retryState`. But if it works in your testing, then it must be fine.

Reply:
Can you elaborate more? That struct has methods with reference, so its state is changed. Is it what you are worried about?

Reply:
Sounds like from our conversation on zoom that this is not an issue. Thanks for helping to explain it to me, I learned something.